All_materials/20230822_Web_Scraping_Financial_Ombudsman_Service/README.md

```diff
@@ -0,0 +1,82 @@
+<br>
+The work is NOT endorsed by the Financial Ombudsman Service.
+
+## How to use the code
+The main code is found in `create_output.py`. The script contains the following steps.
+
+### Constructing the URL link
+
+The construction of the URL link is made using the function:
+
+```
+url_link_from_function = fos_web_scrap.getting_URL_with_date_range(start_date, end_date, config['web']['seach_term']) 
+```
+The inputs of the function are the date range when the complaints were made and the search term use to find the relevant complaints. The parameters can be found in the `config.tmol` file.
+
+### Web scraping
+
+The web scraping is perform using the function:
+```
+complete_list = fos_web_scrap.get_complete_desision_list(url_link_from_function)
+```
+There are hardcoded parameters to extract the relevant html data from the webpage.
+
+### Wrangling and Analysis
+
+The wrangling and analysis are perform using:
+
+```
+_ , _, summary_table = fos_web_scrap.list_of_summary(complete_list)
+```
+The function takes all relevant data from meta data on the html and text from the PDF to a Pandas table.
+
+### Output
+
+The resulting output can be found in `output_table.md`. <br>
+<br>
+For brevity, the main PDF text has been removed before publishing.
+
+
+## Example Output
+The following is example of a table outputted in by the `create_output.py` script. The columns mean the following: <br>
+- Ref – Unique compliant reference 
+- Date – Date of complaint
+- Bank – Financial Institutions which the complaint was made against
+- Scale of Lost – The scale of the largest of sum of money mentioned in the complaint
+- Outcome – If the complaint was upheld
+- CRM – Did the complaint mentioned CRM meaning [Contingent Reimbursement Model Code]( https://www.lendingstandardsboard.org.uk/crm-code/)
+- Sample – A sampled of the complaint from the main body of text
+
+
+| | Ref | Date | Bank | Scale of Lost | Outcome | CRM | Sample |
+|---:|:------------|:--------------------|:---------------------|:----------------|:----------|:------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| 0 | DRN-4009518 | 2023-06-01 00:00:00 | Lloyds Bank PLC | 10^5 | Notupheld | False | omplaint . I know disappointing , I  explain . Was Mr D victim scam ? It dispute Mr D victim scam  initially persuaded invest dating app , introduced third parties applied pressure Mr D part money order access supposed profits . The scammers cut contact Mr D would part money . Did Mr D authorise payments ? In line Payment Services Regulations 2017 ( PSRs ) , Mr D  liable payments  authorise , |
```

Comment:
Is this supposed to start with 'complaint'?

Reply:
no the unique indexing is Ref, the 0,1,2,3 is just pandas's indexing - I'll fix it